### PR TITLE
Added scalafix

### PR DESCRIPTION
**Pull request by olakunlekuye2**
Added scalafix: Scalastyle is removed from the pillar2-frontend project, Scalafix is added to the pillar2-frontend project , All compilation errors are fixed.

### CODE REVIEW

**Comment by ridow-hmrc** [comment]
Can you add a description please?

**Review by ridow-hmrc — APPROVED** [review]

**Review by stipoa — DISMISSED** [review]
Looks good  - changes are very similar to those scalafix made in PIL-1145.

File: .scalafix.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+OrganizeImports {
+  groupedImports = Keep
+  coalesceToWildcardImportThreshold = 3
+}
+
+DisableSyntax {
+  noVars = true
+  noNulls = true
+  noReturns = true
+}
```
Comment (javed-iqbal1) [inline .scalafix.conf:12]: add one empty line at the end of file

File: test/views/rfm/JourneyRecoveryViewSpec.scala [hunk 2]
```diff
@@ -38,16 +39,15 @@ class JourneyRecoveryViewSpec extends ViewSpecBase {
...
-      val linkExists   = view.getElementsByAttributeValue("href", expectedLink).first() != null
+
+      val linkExists = Option(view.getElementsByAttributeValue("href", expectedLink).first()).isDefined
       linkExists mustBe true
 
       view.getElementsByTag("p").text must include(
-        messages("You can go back to")
-          + " "
-          + messages("replace the filing member for a Pillar 2 top-up taxes account to try again")
+        messages("You can go back to") + " " + messages("replace the filing member for a Pillar 2 top-up taxes account to try again")
```
Comment (MoDweik) [inline test/views/rfm/JourneyRecoveryViewSpec.scala:47]: Unrelated to this PR, but messages() is unnecessary here if what's inside is the value and not the key.

Comment (olakunlekuye2) [reply test/views/rfm/JourneyRecoveryViewSpec.scala:47]: Yeah , I'm only making changes suggested by scalafix

File: build.sbt [hunk 3]
```diff
@@ -39,12 +43,9 @@ lazy val root = (project in file("."))
...
     Compile / scalafmtOnCompile := true,
     Test / scalafmtOnCompile := true,
-    scalacOptions ++= Seq(
-      "-feature",
-      "-rootdir",
-      baseDirectory.value.getCanonicalPath,
-      "-Wconf:cat=deprecation:ws,cat=feature:ws,cat=optimizer:ws,src=target/.*:s"
-    ),
+
+    scalacOptions += s"-Wconf:src=${target.value}/scala-${scalaBinaryVersion.value}/routes/.*:s,src=${target.value}/scala-${scalaBinaryVersion.value}/twirl/.*:s",
```
Comment (MoDweik) [inline build.sbt:47]: Has this been suggested by scalafix? Looks different to what's already there. Are the two equivalent?

Comment (olakunlekuye2) [reply build.sbt]: No and I have reverted it.

**Pull request merged**